### PR TITLE
Packages: Move `e2e-tests` to new structure

### LINKED ISSUE #6290: Packages: Move `e2e-tests` to new structure
## Feature Description

This package is a bit different from others as it just contains e2e tests, but would be good to separate these from the test themselves. Similar to how Gutenberg has e2e-tests and e2e-test-utils packages.

* [ ] Move code from `tests/e2e` to `/packages/e2e-tests/src/`
* [ ] Update imports where necessary.  
  Use `@web-stories-wp/...` imports for anything that is already a package, relative imports for everything else.
* [ ] Create `/packages/e2e-tests/package.json` referencing its dependencies
* [ ] Create `/packages/e2e-tests/README.md` with short documentation.
* [ ] Explicitly add `expect-puppeteer` as a dev dependency in `/packages/e2e-tests/package.json` (it's not currently listed as a dep anywhere)

## Additional Context

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by spacedmonkey**
## Relevant Technical Choices

I have not removed many of the packages that may need to be removed, as they maybe used in other parts of the code. I think we should loop back after editor and dashboard are made into packages.

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6290

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -118,10 +118,9 @@
...
     "@testing-library/react-hooks": "^3.7.0",
     "@testing-library/user-event": "^12.6.0",
-    "@web-stories-wp/e2e-test-utils": "*",
+    "@web-stories-wp/e2e-tests": "*",
     "@wordpress/babel-plugin-import-jsx-pragma": "^2.7.0",
     "@wordpress/dependency-extraction-webpack-plugin": "^2.9.0",
-    "@wordpress/e2e-test-utils": "^4.16.1",
     "@wordpress/eslint-plugin": "^7.4.0",
     "@wordpress/jest-console": "^4.0.1",
     "@wordpress/jest-puppeteer-axe": "^3.0.1",
```
Comment (swissspidy) [inline package.json]: These two can be moved to `e2e-tests/package.json`

Comment (swissspidy) [reply package.json]: Same for `@percy/puppeteer`

File: phpcs.xml.dist [hunk 2]
```diff
@@ -80,13 +80,13 @@
...
+	<exclude-pattern>packages/e2e-tests/src/plugins/*</exclude-pattern>
 	<exclude-pattern>*/node_modules/*</exclude-pattern>
 	<exclude-pattern>*/third-party/*</exclude-pattern>
 	<exclude-pattern>*/vendor/*</exclude-pattern>
 	<exclude-pattern>assets/js/*.asset.php</exclude-pattern>
 	<exclude-pattern>includes/polyfills/mbstring.php</exclude-pattern>
 	<exclude-pattern>tests/phpstan/*</exclude-pattern>
 	<exclude-pattern>tests/phpunit/includes/MarkupComparison.php</exclude-pattern>
-	<exclude-pattern>tests/e2e/plugins/*</exclude-pattern>
+	<exclude-pattern>packages/e2e-tests/src/plugins/*</exclude-pattern>
```
Comment (swissspidy) [inline phpcs.xml.dist]: Seems like these are duplicates by accident. Let's remove one

Could also just do `<exclude-pattern>packages/*</exclude-pattern>` to make things simpler.

File: docs/e2e-tests.md [hunk 3]
```diff
@@ -78,6 +78,6 @@ This will launch the browser with the devtools already open. Breakpoints can the
 
 Sometimes one might want to test additional scenarios that aren't possible in a WordPress installation out of the box. The tests setup allows adding some utility plugins that can be activated during E2E tests.
 
-These plugins can be added to `tests/e2e/plugins` and then activated via the WordPress admin.
+These plugins can be added to `packages/e2e-tests/src/plugins` and then activated via the WordPress admin.
 
-Test utilities like `createNewStory()` reside in the `tests/e2e/utils` folder.
+Test utilities like `createNewStory()` reside in the `packages/e2e-tests/src/utils` folder.
```
Comment (swissspidy) [inline docs/e2e-tests.md]:
```suggestion
Test utilities like `createNewStory()` reside in the `packages/e2e-test-utils/src` folder.
```

Comment (spacedmonkey) [reply docs/e2e-tests.md]: Good catch.

File: package.json (hunk 1 above)
Comment (spacedmonkey) [reply package.json]: Are these not used by the karma tests?

Comment (spacedmonkey) [reply package.json]: Do you mean remove these from this file, as they are already in new package.json file, [here](https://github.com/google/web-stories-wp/pull/6305/files#diff-73662dce866d908da409d58c0dd1b3fcef7016ad2125dbb7b01a4172249f5280R12)
[here](https://github.com/google/web-stories-wp/pull/6305/files#diff-73662dce866d908da409d58c0dd1b3fcef7016ad2125dbb7b01a4172249f5280R15)

File: phpcs.xml.dist [hunk 4]
```diff
@@ -80,13 +80,13 @@
 	<!-- Third-party or auto-generated code -->
 	<exclude-pattern>*/build/*</exclude-pattern>
 	<exclude-pattern>*/data/*</exclude-pattern>
-	<exclude-pattern>tests/e2e/plugins/*</exclude-pattern>
+	<exclude-pattern>packages/e2e-tests/src/plugins/*</exclude-pattern>
```
Comment (spacedmonkey) [inline phpcs.xml.dist:83]:
```suggestion
```

File: phpcs.xml.dist (hunk 2 above)
Comment (spacedmonkey) [inline phpcs.xml.dist]:
```suggestion
```

File: package.json (hunk 1 above)
Comment (swissspidy) [reply package.json]: Yes, removed from this file.

And no, they're not used by the Karma tests.

**Comment by swissspidy** [comment]
There's now a merge conflict

File: package.json (hunk 1 above)
Comment (spacedmonkey) [reply package.json]: Thanks for being clear. Removed in https://github.com/google/web-stories-wp/pull/6305/commits/f5f355d1f300c087b7b43bccf3deccd28de5ed96

**Review by swissspidy — APPROVED** [review]

**Pull request merged**